plugin/src/test/java/io/jenkins/plugins/casc/misc/ExportImportRoundTripAbstractTest.java

```diff
@@ -0,0 +1,260 @@
+     * string should be unique to avoid interpreting that it was configured successfully, but it wasn't.
+     * @return the unique string to be in the logs to certify the configuration was done successfully.
+     */
+    public abstract String stringInLogExpected();
+
+    @Before
+    public void saveJenkinsConfig() throws IOException {
+        File jenkinsConfig = new File(r.home, ConfigurationAsCode.DEFAULT_JENKINS_YAML_PATH);
+        if (jenkinsConfig.exists()) {
+            jenkinsConfigContentOld = FileUtils.readFileToString(jenkinsConfig);
+        }
+
+        cascJenkinsConfigPropOld = System.getProperty(ConfigurationAsCode.CASC_JENKINS_CONFIG_PROPERTY, null);
+    }
+
+    /*
+    Annotated with @After is executed before the second step is ended.
+    We need to call it at the end of the second step.
+     */
+    //@After
+    public void restoreJenkinsConfig() throws IOException {
+        if (jenkinsConfigContentOld != null) {
+            writeToFile(jenkinsConfigContentOld, new File(r.home, ConfigurationAsCode.DEFAULT_JENKINS_YAML_PATH).getAbsolutePath());
+        } else {
+            //We remove the file created during the test, it may affect other tests, for example, if the jenkins.yaml
+            //is invalid
+            try {
+                Files.deleteIfExists(Paths.get(new File(r.home, ConfigurationAsCode.DEFAULT_JENKINS_YAML_PATH).toURI()));
+            } catch (FileSystemException e) {
+                //On windows: The process cannot access the file because it is being used by another process. Don't
+                //know where, so far.
+            }
+        }
+
+        if (cascJenkinsConfigPropOld != null) {
+            System.setProperty(ConfigurationAsCode.CASC_JENKINS_CONFIG_PROPERTY, cascJenkinsConfigPropOld);
+        } else {
+            System.clearProperty(ConfigurationAsCode.CASC_JENKINS_CONFIG_PROPERTY);
+        }
+    }
+
+    /**
+     * 1.  Configure the instance with the {@link #configResource()} implemented.
+     * 2.  Check it was configured correctly.
+     * 3.  Export the full Jenkins configuration via Web UI (commented as it fails with maven defaultProperties).
+     * 4.  Export the schema via Web UI (commented out as there is no good validator so far).
+     * 5.  Verify the Jenkins configuration against the schema (commented out as there is no good validator so far).
+     * 6.  Check the Jenkins configuration is valid via Web UI (used the plugin config so far).
+     * 7.  Apply the Jenkins configuration via Web UI (maybe not needed, but we test it as well).
+     * 8.  Write the Jenkins configuration to $JENKINS_ROOT/jenkins.yaml.
+     * 9.  Restart Jenkins.
+     * 10. Check the {@link #stringInLogExpected()} is set during the restart.
+     *
+     * @throws IOException If an exception is thrown managing resources or files.
+     */
+    @Test
+    public void exportImportRoundTrip() throws IOException {
```

Comment:
nit: I've seen similar tests named as `configRoundtrip` in the past. Doesn't it suit here?